<launch>
    <arg name="input_topic" default="/camera/depth/color/points" />
    <!-- arg name="input_topic" default="/velodyne_points" /camera/depth/color/points  -->
    <arg name="no_ground_point_topic" default="/points_no_ground"/>
    <arg name="ground_point_topic" default="/points_ground"/>
    <arg name="all_points_topic" default="/all_points"/>

    <!-- clip the points above sensor_height+clip_height -->
    <arg name="clip_height" default="4.0"/>
    <!-- the height of lidar position  0.77-->
    <arg name="sensor_height" default="0.157"/>
    <!-- the min distance to be remove 1-->
    <arg name="min_distance" default="0.1"/>
    <!-- the max distance to be remove 100-->
    <arg name="max_distance" default="4.0"/>
    <!-- we use 32 lidar -->
    <arg name="sensor_model" default="32"/>
    <!-- Num of Iteration 3-->
    <arg name="num_iter" default="2"/>
    <!-- Num of LPR 20-->
    <arg name="num_lpr" default="20"/>
    <!-- Seeds Threshold 1.2-->
    <arg name="th_seeds" default="0.2"/>
    <!-- Distance Threshold 0.3-->
    <arg name="th_dist" default="0.03"/>


    <node pkg="plane_ground_filter" type="plane_ground_filter_node" name="plane_ground_filter_node" output="screen">
        <rosparam param="Extri_T_baselink_camera">[0.457, 0., 0.157, 0.509, -0.491, 0.491, -0.509]</rosparam>
        <param name="input_topic" value="$(arg input_topic)"/>
        <param name="no_ground_point_topic" value="$(arg no_ground_point_topic)"/>
        <param name="ground_point_topic" value="$(arg ground_point_topic)"/>
        <param name="all_points_topic" value="$(arg all_points_topic)"/>

        <param name="sensor_height" value="$(arg sensor_height)" />
        <param name="clip_height" value="$(arg clip_height)" />
        <param name="min_distance" value="$(arg min_distance)" />
        <param name="max_distance" value="$(arg max_distance)" />
        <param name="sensor_model" value="$(arg sensor_model)" />
        <param name="num_iter" value="$(arg num_iter)" />
        <param name="num_lpr" value="$(arg num_lpr)" />
        <param name="th_seeds" value="$(arg th_seeds)" />
        <param name="th_dist" value="$(arg th_dist)" />
    </node>
    <!-- 启动rviz -->
    <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d /home/lxf/lxf_package/lxf_work_deep_camera/ws_filter/src/plane_fit_ground_filter-master/plane_ground_filter_core.rviz"/> -->

</launch>
